src/HotelManagementSystem/FrontDesk/HotelManagementSystem.FrontDesk.Client/src/app/pages/protected/booking-details/booking-details.component.ts

```diff
@@ -1,10 +1,51 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { BookingService } from '../services/booking.service';
+import { Booking } from '../models/booking';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { ReasonModalComponent } from '../reason-modal/reason-modal.component';
+import { Room } from '../models/room';
+import { RoomService } from '../services/room.service';
 
 @Component({
   selector: 'app-booking-details',
   templateUrl: './booking-details.component.html',
-  styleUrl: './booking-details.component.css'
+  styleUrls: ['./booking-details.component.css']
 })
-export class BookingDetailsComponent {
+export class BookingDetailsComponent implements OnInit {
+  booking: Booking;
+  rooms: Room[];
 
-}
+  constructor(private route: ActivatedRoute, private bookingService: BookingService, private roomService: RoomService, private modalService: NgbModal) { }
+
+  ngOnInit(): void {
+    this.getBookingDetails();
+  }
+
+  getBookingDetails(): void {
+    const id = +this.route.snapshot.paramMap.get('id');
+    this.bookingService.getBooking(id)
+      .subscribe(booking => {
+        this.booking = booking;
+        this.loadAvailableRooms(booking.roomType.id);
```

Comment:
Only load the available rooms, when the user clicks on Accept or Decline button.